File: IMFS.Web.Models/User/UserModel.cs
using IMFS.Web.Models.Misc;
using System.Collections.Generic;

namespace IMFS.Web.Models.User
{
    public class UserModel
    {
        public string Id { get; set; }
        public string Title { get; set; }
        public string FullName { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string UserName { get; set; }
        public string JobTitle { get; set; }
        public string Email { get; set; }
        public int? DefaultFinanceType { get; set; }
        public int? DefaultFinanceDuration { get; set; }
        public string DefaultFinanceFrequency { get; set; }
        public string DefaultFinanceTypeName { get; set; }
        public int DefaultFinanceDurationName { get; set; }

        public string PhoneNumber { get; set; }

        public List<LookupModel> Roles { get; set; }


        //Editable by IMStaff
        public bool Active { get; set; }
        public bool ExcludeGST { get; set; }
        public string RoleId { get; set; }
        public string RoleName { get; set; }

        public string CustomerNumber { get; set; }
        public string CustomerName { get; set; }


    }

    public class UserSearchModel
    {

        public string FullName { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }

        public string JobTitle { get; set; }
        public string Email { get; set; }
        public int? DefaultFinanceType { get; set; }
        public int? DefaultFinanceDuration { get; set; }
        public string DefaultFinanceFrequency { get; set; }
        public string DefaultFinanceTypeName { get; set; }
        public int DefaultFinanceDurationName { get; set; }

        public string PhoneNumber { get; set; }


        public bool Active { get; set; }

        public string RoleId { get; set; }
        public string RoleName { get; set; }

        public string CustomerNumber { get; set; }
        public string CustomerName { get; set; }


    }

    public class AutoCreateUserTemplateModel
    {
        public string IngramLogoHTML { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string Email { get; set; }
        public string CustomerNumber { get; set; }
        public string CustomerName { get; set; }
        public string Role { get; set; }
        public string IMFSHostLink { get; set; }
        public string IMFSUserLink { get; set; }
    }

    public class CurrentUserInfo
    {
        public string UserId { get; set; }
        public string UserName { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string Email { get; set; }
        public string CustomerNumber { get; set; }
        public string CompanyName { get; set; }
        public string Phone { get; set; }
        public string Role { get; set; }
    }
}

File: IMFS-Web-FrontEnd/src/app/models/user/user.model.ts

export class UserResponseModel {
    id: string;
    title: string;
    firstName: string;
    lastName: string;
    fullName: string;
    userName: string;
    jobTitle: string;
    email: string;
    phoneNumber: string;
    customerNumber: string;
    customerName: string;
    defaultFinanceType: number | null;
    defaultFinanceDuration: number | null;
    defaultFinanceFrequency: string;

    defaultFinanceTypeName: string;
    defaultFinanceDurationName: string;
    active: boolean;
    excludeGST: boolean;

    roleId: string;
    roleName: string;

}

export class UserUpdateModel {
    id: string;
    title: string;
    firstName: string;
    lastName: string;
    userName: string;
    jobTitle: string;
    email: string;
    phoneNumber: string;
    customerNumber: string;
    customerName: string;
    defaultFinanceType: number | null;
    defaultFinanceDuration: number | null;
    defaultFinanceFrequency: string;

    active: boolean;
    excludeGST: boolean;
    role: string;
}

export class UserSearchModel {
    fullName: string;
    firstName: string;
    lastName: string;
    email: string;
    active: boolean;
    customerNumber: string;
    customerName: string;
    jobTitle: string;
}

export class CurrentUserInfo {
    userId: string;
    userName: string;
    firstName: string;
    lastName: string;
    email: string;
    customerNumber: string;
    companyName: string;
    phone: string;
    role: string;
}
